File: app/src/main/java/com/zerostress/manager/models/ChatMessage.java
package com.zerostress.manager.models;

public class ChatMessage {
    private String id;
    private String senderId;
    private String senderName;
    private String text;
    private long timestamp;
    private boolean deleted;

    public ChatMessage() {}

    public ChatMessage(String senderId, String senderName, String text) {
        this.senderId = senderId;
        this.senderName = senderName;
        this.text = text;
        this.timestamp = System.currentTimeMillis();
        this.deleted = false;
    }

    public String getId() { return id; }
    public String getSenderId() { return senderId; }
    public String getSenderName() { return senderName; }
    public String getText() { return text; }
    public long getTimestamp() { return timestamp; }
    public boolean isDeleted() { return deleted; }

    public void setId(String id) { this.id = id; }
    public void setSenderId(String senderId) { this.senderId = senderId; }
    public void setSenderName(String senderName) { this.senderName = senderName; }
    public void setText(String text) { this.text = text; }
    public void setTimestamp(long timestamp) { this.timestamp = timestamp; }
    public void setDeleted(boolean deleted) { this.deleted = deleted; }
}

File: app/src/main/java/com/zerostress/manager/models/ChatMessage.kt
package com.zerostress.manager.models

import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
data class ChatMessage(
    var id: String = "",
    var senderId: String = "",
    var senderName: String = "",
    var text: String = "",
    var timestamp: Long = 0L,
    var deleted: Boolean = false,
    var mentions: List<String> = emptyList()
) {
    companion object {
        fun create(senderId: String, senderName: String, text: String): ChatMessage =
            ChatMessage(
                senderId = senderId,
                senderName = senderName,
                text = text,
                timestamp = System.currentTimeMillis(),
                deleted = false
            )
    }
}
